bokeh/application/handlers/handler.py

```diff
@@ -20,6 +21,13 @@ def static_path(self):
         else:
             return self._static
 
+    def template(self):
+        """Returns an app-specific custom Jinja template, if applicable."""
+        if self.failed:
+            return None
+        else:
+            return self._template
```

Comment:
Instead of this, modify_document could set `doc.template` if appropriate; it could also raise an error if doc.template is set already. Possibly the error wouldn't be quite as clear, since we wouldn't know inside modify_document which earlier handler had already set the template.